src/ClimaLSM.jl

```diff
@@ -103,22 +104,89 @@ Returns the component names of the `land` model, by calling
 """
 land_components(land::AbstractLandModel) = propertynames(land)
 
+function prognostic_vars(land::AbstractLandModel)
+    components = land_components(land)
+    prognostic_list = map(components) do model
+        prognostic_vars(getproperty(land, model))
+    end
+    return NamedTuple{components}(prognostic_list)
+end
+
+function prognostic_types(land::AbstractLandModel)
+    components = land_components(land)
+    prognostic_list = map(components) do model
+        prognostic_types(getproperty(land, model))
+    end
+    return NamedTuple{components}(prognostic_list)
+end
+
+function auxiliary_vars(land::AbstractLandModel)
+    components = land_components(land)
+    auxiliary_list = map(components) do model
+        auxiliary_vars(getproperty(land, model))
+    end
+    return NamedTuple{(components..., :interactions)}((
+        auxiliary_list...,
+        interactions_vars(land),
+    ))
+end
+
+function auxiliary_types(land::AbstractLandModel)
+    components = land_components(land)
+    auxiliary_list = map(components) do model
+        auxiliary_types(getproperty(land, model))
+    end
+    return NamedTuple{(components..., :interactions)}((
+        auxiliary_list...,
+        interactions_types(land),
+    ))
+end
+
 """
-    initialize_interactions(land::AbstractLandModel) end
+   interaction_vars(m::AbstractModel)
 
-Initializes interaction variables, which are a type of auxiliary
-variable, to empty objects of the correct type
-for the model. 
+Returns the interaction variable symbols for the model in the form of a tuple.
+"""
+interactions_vars(m::AbstractLandModel) = ()
+
+"""
+   interaction_types(m::AbstractModel)
+
+Returns the shared interaction variable types for the model in the form of a tuple.
+"""
+interactions_types(m::AbstractLandModel) = ()
+
+"""
+   interaction_spaces(m::AbstractModel)
 
-This function should be called during `initialize_auxiliary`,
-and the method will change depending on the type of land model,
-and potentially due to the type of the component models. 
+Returns the symbol of a model on the interaction space in the form of a tuple.
 
-This is a stub which is extended to concrete LSM models.
+For example, for a boundary interaction term, one would specify a model with a
+domain consisting of the boundary space. Most commonly, these are surface models.
+"""
+interactions_spaces(m::AbstractLandModel) = ()
```

Comment:
This is the only piece I'm a little grumpy about. It associates interactions variables with a submodel so that they can be initialized on the correct space. The downside is that these interactions don't really belong to a given submodel. So perhaps there is a better way to specify what boundaries these interactions occur on, but this is fast and easy.

Please suggest any language that would make this function & its docstring more clear.

Reply:
Another idea: let's call it `interactions_domain`, b/c that is what we are really talking about, and reserve "spaces" for a future use case where we need to distinguish between centers and faces?

and we can also note that component specific variables do not require this information, because by definition they are defined at each point in the coordinate field of their component's domain. Interaction variables are more flexible, so we had to introduce this.

Reply:
> let's call it `interactions_domain`

Sounds good - that makes it more consistent with our use of "domain" in LSM.

> and we can also note that component specific variables do not require this information

I'll add a note to the docstring :)

>Maybe instead of associating them with a component, we can associate them with a part of the LSMDomain? This is split into subsurface and surface domains (domain.subsurface, domain.surface). It would be more in line with them being shared to pass in :surface or :subsurface rather than e.g. :soil?

I like that, need to think about the easiest way to implement that. Mostly, it is then associating those tags with component models.

Reply:
> It would be more in line with them being shared to pass in :surface or :subsurface rather than e.g. :soil?

I think this will change a few other things, so I will add this in a separate PR since this is getting quite long already.